lib/ansible/plugins/callback/default.py

```diff
@@ -40,6 +40,10 @@ def __init__(self):
         self._last_task_banner = None
         super(CallbackModule, self).__init__()
 
+    def _dump_results(self, result, **kwargs):
+        result = { key: result[key] for key in result if key not in ['deprecation', 'warnings'] }
+        return super(CallbackModule, self)._dump_results(result, **kwargs)
+
```

Comment:
`warnings` and `deprecations` keys are handled in [`_handle_warnings`](https://github.com/pilou-/ansible/blob/display_warnings_failure/lib/ansible/plugins/callback/__init__.py#L100) method which is called after the call to [`_dump_results`](https://github.com/pilou-/ansible/blob/display_warnings_failure/lib/ansible/plugins/callback/default.py#L75): if these keys were deleted after they are handled, they will be displayed twice in some cases:
* in case of failure:
    * first in the error message ([`v2_runner_on_failed`](https://github.com/pilou-/ansible/blob/display_warnings_failure/lib/ansible/plugins/callback/default.py#L66))
    * in the message prefixed by `WARNING` (`_handle_warnings` is [called](https://github.com/pilou-/ansible/blob/display_warnings_failure/lib/ansible/plugins/callback/default.py#L75) at the end of `v2_runner_on_failed`)
* in case of success when the debug callback is used (`_handle_warnings` is called last too)

Either displaying warnings twice is fine, or we should display warnings then delete the keys, both before the error message. What do you prefer ?

Note that:
- I kept `result._result` because `v2_runner_on_failed` doesn't modify `result._result`directly (instead a copy is created, modified and returned by [`_dump_results`](https://github.com/pilou-/ansible/blob/display_warnings_failure/lib/ansible/plugins/callback/__init__.py#L83)).
- In `v2_runner_on_failed`, I added the call to `_handle_warnings` after the output of the error because `v2_runner_on_ok do that.

Thank you for your review !